data_module.py

```diff
@@ -2,8 +2,11 @@
 from pathlib import Path
 from torch.utils.data import DataLoader
 from ThermalClassifier.datasets.download_dataset import download_dataset
```

Comment:
I implemented download folder in SoiUtils, Its a safer version than download_dataset function you are using here.